### ISSUE TITLE
CodeLens "Debug" appears on wrong document

### ISSUE THREAD

**Issue by seanpoulter** [jest-community/vscode-jest#178]
### Environment

1. `node -v`: v8.9.1
2. `npm -v`: 5.5.1
4. `npm ls react-scripts` (if you haven’t ejected): N/A
1. Operating system: Windows 7 x64
1. VS Code: 1.18.1
1. Jest extension: 4.5.0

### Steps to Reproduce

1. Clone the repo:
    ```
    git clone https://github.com/seanpoulter/vscode-jest-178
    ```
1. Open the repo in Visual Studio Code
1. Open `index.js` and `index.test.js`
1. Wait for the CodeLens "Debug" to appear in `index.test.js`
1. Switch to another window

### Expected Behavior

The "Debug" should **only** appear on the appropriate document.

### Actual Behavior

"Debug" appears on the wrong document. Here's a GIF of it appearing correctly on `index.test.js` then incorrectly on `index.js`.

![2017-12-01 -- vscode-jest 178 - debug appears in wrong window](https://user-images.githubusercontent.com/2585460/33511492-6b90e3c6-d6e9-11e7-82af-99b9a726d4fa.gif)

**Comment by bogdan-calapod** [comment]
I would say this is related to #175 ? The feature requested is caused by exactly this bug - the debug button shows up in the inappropiate document and flips around scrolling in VSCode

**Comment by bogdan-calapod** [comment]
After some investigation I noticed that the "Debug" thing shows up correctly if you toggle side-by-side view of the `.js` file and the `.test.js` file

**Comment by seanpoulter** [comment]
Yep, @bogdan-calapod - the behavior is described in #175. Since #175 is about adding a setting to disable the "Debug" CodeLens I thought it'd be easier to track this as a separate issue.

**Comment by goncharov** [comment]
The current implementation of codelens uses the common list of [decorations](https://github.com/orta/vscode-jest/blob/master/src/CodeLens.ts#L18) which updates implicitly by change active text editor [only if a new document is jest test](https://github.com/orta/vscode-jest/blob/master/src/JestExt.ts#L349). If the document in new editor is not Jest test it will not reset previous decorations and will display old ones. This is why lenses display in wrong files. It also reproduces for side-by-side view

**Comment by seanpoulter** [comment]
We need to confirm if this was fixed by #186 and ideally test it.

**Comment by orta** [comment]
#186 really only adds the ability to remove all of them - @goncharov's answer would probably fix it 👍
